01-html/01-koajs.html

```diff
@@ -1,4 +1,671 @@
+        <li><a href="context">context</a></li>
+        <li><a href="request">request</a></li>
+        <li><a href="response">response</a></li>
+        <li><a href="links">links</a></li>
+      </ul>
+    </nav>
+  </header>
+  <hgroup>
+    <h2>Introduction</h2>
+      <p>Koa is a new web framework designed by the team behind Express, which aims to be a smaller, more expressive, and more robust foundation for web applications and APIs. Through leveraging generators Koa allows you to ditch callbacks and greatly increase error-handling. Koa does not bundle any middleware within core, and provides an elegant suite of methods that make writing servers fast and enjoyable.</p>
+    <h2>Installation</h2>
+      <p>Koa works out of the box with recent versions of Node. To use Koa with 0.11.x
+        you must use the <em>--harmony</em> or <em>--harmony-generators</em> flag.</p>
+      <code>$npm install koa<br>$node app.js</code>
+    <h2>Application</h2>
+      <p>A Koa application is an object containing an array of middleware generator functions which are composed and executed in a stack-like manner upon request. Koa is similar to many other middleware systems that you may have encountered such as Ruby's Rack, Connect, and so on - however a key design decision was made to provide high level <q>sugar</q> at the otherwise low-level middleware layer. This improves interoperability, robustness, and makes writing middleware much more enjoyable.</p>
+      <p>This includes methods for common tasks like content-negotiation, cache freshness, proxy support, and redirection among others. Despite supplying a reasonably large number of helpful methods Koa maintains a small footprint, as no middleware are bundled.</p>
+      <p>The obligatory hello world application:</p>
+        <code>var koa = require('koa');<br>var app = koa();
+          <br>app.use(function *(){
+          <br>this.body = 'Hello World';
+          <br>});
+          <br>app.listen(3000);</code>
+      <h3>Cascading</h3>
+        <p>Koa middleware cascade in a more traditional way as you may be used to with similar tools - this was previously difficult to make user friendly with node's use of callbacks. However with generators we can achieve <q>true</q> middleware. Contrasting Connect's implementation which simply passes control through series of functions until one returns, Koa yields <q>downstream</q>, then control flows back <q>upstream</q>.</p>
+        <p>The following example responds with <q>Hello World</q>, however first the request flows through the <em>x-response-time</em> and <em>logging</em> middleware to mark when the request started, then continue to yield control through the response middleware. When a middleware invokes <em>yield next</em> the function suspends and passes control to the next middleware defined. After there are no more middleware to execute downstream, the stack will unwind and each middleware is resumed to perform its upstream behaviour.</p>
```

Comment:
`<q>` is generally reserved for a specific quote in which you might want to cite a source, such as a news or encyclopedia article, or if you wanted to style the contents in a particular way. For the vast majority of the time, simply quotation marks is sufficient.